editors/sc-ide/core/sc_server.cpp

```diff
@@ -392,9 +411,29 @@ void ScServer::onScLangReponse( const QString & selector, const QString & data )
 	else if (selector == startDumpOSCSelector) {
         mActions[DumpOSC]->setChecked(true);
     }
-	else if (selector == stopDumpOSCSelector) {
+    else if (selector == stopDumpOSCSelector) {
         mActions[DumpOSC]->setChecked(false);
     }
+	else if (selector == recordingDurationSelector) {
+        bool ok;
+        float duration = data.mid(1, data.size() - 2).toFloat(&ok);
+        if (ok) {
+            mRecordTime = (int) duration;
+            updateRecordingAction();
+        }
+    }
+    else if (selector == startRecordingSelector) {
+        setRecording(true);
+    }
+    else if (selector == startRecordingSelector) {
+        setRecording(true);
+    }
+    else if (selector == pauseRecordingSelector) {
+        pauseRecording(true);
+    }
+    else if (selector == stopRecordingSelector) {
+        setRecording(false);
+    }
```

Comment:
it might be reasonable to introduce a `Q_PROPERTY` (http://doc.qt.io/qt-5/properties.html) with an `enum RecordingState { stopped, running, paused}` and propagate all changes via a `recordingStateChanged` notification (http://doc.qt.io/qt-5/signalsandslots.html#signals-and-slots).